### ISSUE TITLE
您好，请问能否分享一下做Visualization部分的代码？

### ISSUE THREAD

**Issue by zongzhesun** [raoyongming/CAL#7]
您好，请问能否分享一下做Visualization部分的代码？我使用了GradCAM来进行可视化，可一直存在BUG，可视化出来的结果很差，所以想请教一下您是怎么做可视化的？十分期待您的回复，谢谢！

**Comment by raoyongming** [comment]
Hi, we first normalize the attention map to [0, 1]. Then, we use the follow code to visualize the attention map:
```python
def generate_heatmap(attention_maps):
    heat_attention_maps = []
    heat_attention_maps.append(attention_maps[:, 0, ...])  # R
    heat_attention_maps.append(attention_maps[:, 0, ...] * (attention_maps[:, 0, ...] < 0.5).float() + \
                               (1. - attention_maps[:, 0, ...]) * (attention_maps[:, 0, ...] >= 0.5).float())  # G
    heat_attention_maps.append(1. - attention_maps[:, 0, ...])  # B
    return torch.stack(heat_attention_maps, dim=1)
```

**Comment by zongzhesun** [comment]
Thank you a lot, I'll have a try.

**Comment by zongzhesun** [comment]
您好，我尝试使用您提供的方法做可视化，可还是存在一些实现上的问题。不知您能否分享一个包含具体实现可视化过程的完整代码呢？十分感谢！
